lib/std/zig/CrossTarget.zig

```diff
@@ -45,6 +45,9 @@ dynamic_linker: DynamicLinker = DynamicLinker{},
 /// `null` means default for the cpu/arch/os combo.
 ofmt: ?Target.ObjectFormat = null,
 
+/// `null` means to dynamically get the page size or use the platform default.
```

Comment:
1. Kernel support varies alot, so unless this is checked, this will remain a footgun + introduce maintenance churn.
2. This is also missing the hardware bits for freestanding.
3. Can you quickly show the docs showing that the page size is part of the object format on Windows and Mac/Darwin? One could at least claim on Posix, that writing it to aux vector should "be sufficient to be checked".

Reply:
1. Majority of kernels should support the auxiliary vector. Earliest I can find from a quick search dates back to 2012 (https://lwn.net/Articles/519085/). If there's a Linux system which doesn't then we can do an ABI check.
2. Freestanding should just work as it would fall back to the old method.
3. I'm not sure what you mean by that.

Reply:
Another point on 1, a spec from 2002 mentions it (https://refspecs.linuxfoundation.org/LSB_1.3.0/IA64/spec/auxiliaryvector.html). Suffice to say, I think only in extremely rare cases would someone run into such an incompatibility.

Reply:
> Majority of kernels should support the auxiliary vector.

Do you know how it works on Windows?

> I'm not sure what you mean by that.

I am not sure, if MacOS or Windows allows an application to specify the auxiliary vector at comptime and what guarantee on the stability of the auxiliary vector MacOS and Windows provide (backwards and forwards). This is relevant for a decision, if we should support it at all and/or what information should be put into the doc comments regarding portability guarantees.

Reply:
Auxiliary vector is a Linux feature only afaik because it is a part of ELF and macOS uses Dylib, Windows uses PE.

Reply:
I don't have any clue as to how to add Windows or macOS support.